### DISCUSSION TITLE
Thread collecting various ideas for Dasharo Tools Suite improvements

### DISCUSSION THREAD

**Discussion by macpijan** [Dasharo/dasharo-issues#361] [Ideas]
- [ ] Use external library for version parsing and comparison. RC versions are not parsed correctly right now, for example. Examples:
    - https://github.com/fsaintjacques/semver-tool
    - https://github.com/Ariel-Rodriguez/sh-semversion-2
   
- [ ] Place scripts into a seperate repositories. Add pre-commit and shellcheck to enforce style guidelines and check for common errors.
- [ ] Improve UI. Suggestions:
    - https://mrchromebox.tech/#fwscript
    - https://github.com/MrChromebox/scripts
- [ ] Use actual programming language, rather than shell scripts? Researched some TUI libraries. This one is very active with lots of examples:
    - https://github.com/charmbracelet/bubbletea/tree/master/examples
- [ ] More testing on different levels
    - https://github.com/bats-core/bats-core

**Comment by macpijan** [comment]
Yesterday at night I created repository, experimenting a bit with v2 menu: https://github.com/Dasharo/dts/tree/init

![2023-03-02T12:27:53,539756028+01:00](https://user-images.githubusercontent.com/19353650/222416296-2f774d93-32ef-4465-85f6-47f13085de71.png)

**Reply by pietrushnic** [reply]
I love it. What about firmware update? I guess we can change menu items when detected firmware is already Dasharo.
- I think we should also give credits to MrChromebox for inspiration. (inspired by or sth)
- maybe we should have about the application or sth there

**Reply by pietrushnic** [reply]
Let's remember to promote this discusison and other important topics on DUG and vPub.

**Reply by macpijan** [reply]
> I love it. What about firmware update? I guess we can change menu items when detected firmware is already Dasharo.

Exactly. I though about it, and set it in a way that the same option will say `update` once Dasharo is installed, or `install` when it is not.

**Reply by macpijan** [reply]
> I think we should also give credits to MrChromebox for inspiration. (inspired by or sth)

Yes - of course

**Comment by pietrushnic** [comment]
Some suggestions from me were already submitted in this issue: https://github.com/Dasharo/dasharo-issues/issues/352

**Comment by macpijan** [comment]
It should be simple to find the source of the currently running DTS scripts. If someone wants inspect, or fix something. Observation from Matrix reported by Marek: https://matrix.to/#/!rsKWMJGPMsyPTTjXuh:matrix.org/$H7kIfXP7rFA__-L_-gZzQQErW1dFy2Fph8hDlARgP0w?via=matrix.org&via=invisiblethingslab.com&via=matrix.optoutpod.com

**Reply by marmarek** [reply]
I did find the right repository, but was looking at wrong branch (main vs develop).

**Reply by macpijan** [reply]
@TomaszAIR So far, putting the -dev version in the iPXE menu cause a lot of confusion.
